actionpack/lib/action_controller/metal/instrumentation.rb

```diff
@@ -71,8 +71,8 @@ def redirect_to(*args)
   private
 
     # A hook invoked every time a before callback is halted.
-    def halted_callback_hook(filter)
-      ActiveSupport::Notifications.instrument("halted_callback.action_controller", :filter => filter)
+    def halted_callback_hook(callback)
+      ActiveSupport::Notifications.instrument("halted_callback.action_controller", callback: callback)
```

Comment:
https://travis-ci.org/rails/rails/jobs/132127369.  looks like i missed wherever `payload[:filter]` is being called.  This change is obviously more invasive than the docs, so I leave that up to core if it is worth it.